data-model/src/main/scala/za/co/absa/enceladus/model/properties/PropertyDefinition.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.model.properties
+
+import java.time.ZonedDateTime
+
+import com.fasterxml.jackson.databind.node.{ArrayNode, ObjectNode}
+import za.co.absa.enceladus.model.menas.MenasReference
+import za.co.absa.enceladus.model.menas.audit.{AuditFieldName, AuditTrailChange, AuditTrailEntry, Auditable}
+import za.co.absa.enceladus.model.properties.essentiality.{Essentiality, Mandatory, Optional}
+import za.co.absa.enceladus.model.properties.propertyType.PropertyType
+import za.co.absa.enceladus.model.versionedModel.VersionedModel
+
+case class PropertyDefinition(name: String,
+                              version: Int = 1,
+                              description: Option[String] = None,
+
+                              propertyType: PropertyType,
+                              putIntoInfoFile: Boolean = false,
+                              essentiality: Essentiality = Optional(),
```

Comment:
I know that this is a result of a compromise, but it seems way too complicated when compared to a simple boolean `required`. Or two (`required`, `recommended`?). If we even need recommended...